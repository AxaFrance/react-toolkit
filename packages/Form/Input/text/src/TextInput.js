import React from 'react';
import PropTypes from 'prop-types';

import {
  Field,
  HelpMessage,
  FieldInput,
  FieldConstants as Constants,
  FormClassManager,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import Text from './Text';

const propTypes = {
  ...Constants.propTypes,
  value: PropTypes.string,
  placeholder: PropTypes.string,
};

const defaultProps = {
  ...Constants.defaultProps,
  placeholder: null,
  value: '',
};

const TextInput = props => {
  const {
    classModifier,
    message,
    children,
    helpMessage,
    id,
    disabled,
    name,
    classNameContainerLabel,
    classNameContainerInput,
    value,
    onChange,
    readOnly,
    placeholder,
    label,
    messageType,
    isVisible,
    className,
    forceDisplayMessage,
    maxLength,
    minLength,
    ...otherProps
  } = props;
  const inputClassModifier = FormClassManager.getInputClassModifier(
    classModifier,
    children
  );
  const inputFieldClassModifier = FormClassManager.getFieldInputClassModifier(
    classModifier,
    disabled
  );
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      helpMessage={helpMessage}
      message={message}
      childrenRight={children}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      id={inputId}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        helpMessage={helpMessage}
        message={message}
        messageType={messageType}
        className="af-form__text"
        classModifier={inputFieldClassModifier}>
        <Text
          name={name}
          id={inputId}
          value={value}
          onChange={onChange}
          readOnly={readOnly}
          disabled={disabled}
          placeholder={placeholder}
          classModifier={inputClassModifier}
          maxLength={maxLength}
          minLength={minLength}
          {...otherProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

TextInput.propTypes = propTypes;
TextInput.defaultProps = defaultProps;

export default TextInput;
