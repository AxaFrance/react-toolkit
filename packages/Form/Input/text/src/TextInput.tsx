import React, { ComponentProps } from 'react';
import {
  Field,
  HelpMessage,
  FieldInput,
  withInputClassModifier,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import { ComponentType } from 'enzyme';
import Text from './Text';

type Props = ComponentProps<typeof Field> &
  ComponentProps<typeof Text> & {
    inputFieldClassModifier: string;
    inputClassModifier: string;
    helpMessage?: string;
    children?: ReactNode;
  };

const TextInput = ({
  classModifier,
  message,
  children,
  helpMessage,
  id,
  classNameContainerLabel,
  classNameContainerInput,
  onChange,
  label,
  messageType,
  isVisible,
  className,
  forceDisplayMessage,
  inputFieldClassModifier,
  inputClassModifier,
  ...inputTextProps
}: Props) => {
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      id={inputId}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        className="af-form__text"
        classModifier={inputFieldClassModifier}>
        <Text
          id={inputId}
          onChange={onChange}
          classModifier={inputClassModifier}
          {...inputTextProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

const enhanced = withInputClassModifier(TextInput);
export default enhanced;
