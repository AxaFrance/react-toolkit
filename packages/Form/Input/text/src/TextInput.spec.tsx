import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe('<TextInput>', () => {
  it('renders SwitchInput correctly', () => {
    const tree = renderer
      .create(
        <TextInput label="Image *" id="iddateinput" value="ici le text" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
