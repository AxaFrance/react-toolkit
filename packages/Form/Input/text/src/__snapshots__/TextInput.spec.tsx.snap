// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> renders SwitchInput correctly 1`] = `
<div
  className="row af-form__group"
>
  <div
    className="col-md-2"
  >
    <label
      className="af-form__group-label"
      htmlFor="iddateinput"
    >
      Image *
    </label>
  </div>
  <div
    className="col-md-10"
  >
    <div
      className="af-form__text af-form__text--undefined"
    >
      <input
        className="af-form__input-text af-form__input-text--undefined"
        id="iddateinput"
        onChange={[Function]}
        type="text"
        value="ici le text"
      />
    </div>
    <small
      className="af-form__help"
    />
  </div>
</div>
`;
